content/articles/core-concepts-of-django-to-master/index.md

```diff
@@ -0,0 +1,166 @@
+
+### Keys Take Away
+At the end of this article, you will be able to do the following;
+1. Setup Django project and configure it
+2. Have core understanding of how routers work
+3. Know what we mean by MVT in Django world
+4. Understand Django Admin Interface
+5. Be able to authenticate users 
+6. Know what is relational database management system
+
+Let's Begin!!!!
+
+What is Django Framework?
+
+"Django is a free, open source web framework written in the Python programming language". A “web framework” is software that abstracts away many of the common challenges related to building a website, such as connecting to a database, handling security, user accounts and much more. Django uses a python package called pip and this come in handy having installed Python. 
+Django  is used to configure the server side of most website applications as used by Shopify, Google, Pinterest etc. This web framework allows developers out there to develop the backend by actually making ties with frontend of aplications through the use of it concepts, as we want to discuss in this article and databases in a real  world applications. Been a django developer with few years of experience, I have come to conclude on the these concepts as the backbone of Django which you cannot escape if truly you  want to be a succesful Django developer. Below is the bulletin of concepts to be mastered whether you are just starting out learning the framework or you have been into it for a while.
+
+1. Django Project Structure
+2. How Route Works In Web Applications using Django
+3. Understanding Of Model View Template (MVT)
+4. Dajngo Admin Interface Usage
+5. Django User Authentication System
+6. Basics Of Relational Database Management System
+
+<!-- PART ONE -->
+DJANGO PROJECT STRUCTURE
+
+Let me begin by congratulating you for choosing Django as favorite web framework. Now how do we set up Django project in real world?
+Least I forget, Django uses a virtual enviroment. What is virtual enviroment?
+Virtual Enviroment is just a file/folder that contains all the Django dependencies installed from Python pip command which are neccessary  for the framework. All these dependecies are shipped out together with the project when it is time to deploy to real world by freezing them into a file called requirements.txt with command below.
+
+* pip freeze > requirements.txt
+
+How To Set Up Django Project 
+
+Firstly, download and configure any code editor of your choice be it VSCODE, SUBLIME, ATOM e.t.c. Then head to python.org to download latest python to your machine which is Python3.9 currently. Python uses a command known as "pip" to install any of it package. For this reason, you will be using this alot whenever you are to install or update any Python package. Now its time to open your editor, lets get our hands dirty!! 
+Open the terminal of the editor at the top center and type the following commands: 
+
+* ls   ----> this print out files at the current path you are.
```

Comment:
Tip:
You can format these commands as code so it is easy for the reader to copy and past them. For example:

```python
cd django-project # changing directory to the folder
```

Reply:
Okay sir! Already working on the corrections.